mobile/src/main/java/ch/epfl/sdp/mobile/application/chess/implementation/NormalizedBoard.kt

```diff
@@ -62,4 +66,8 @@ class DenormalizedBoardDecorator(
     val result = board.set(player.normalize(position), piece?.denormalize())
     return DenormalizedBoardDecorator(player, result)
   }
+
+  override fun iterator(): Iterator<Pair<Position, Piece<Color>>> {
+    return board.asSequence().map { (pos, piece) -> pos to piece.normalize() }.iterator()
```

Comment:
The normalisation / denormalisation processes are actually identical since it only consists in a vertical symmetry for the black player. However I'll see if / how I can rename things to make them more consistent with the name of the decorator.